msm/src/mvlookup.rs

```diff
@@ -0,0 +1,403 @@
+                    for f_i in f.iter() {
+                        let Lookup {
+                            numerator,
+                            table_id: _,
+                            value: _,
+                        } = &f_i[j as usize];
+                        row_acc += *numerator * denominators[denominator_index];
+                        denominator_index += 1;
+                    }
+                    // We process t now
+                    let Lookup {
+                        numerator,
+                        table_id: _,
+                        value: _,
+                    } = &t[j as usize];
+                    row_acc += *numerator * denominators[denominator_index];
+                    denominator_index += 1;
+                    evals.push(row_acc)
+                }
+                // FIXME: only one lookup terms for now
+                evals
+            };
+
+            // FIXME: only one lookup terms for now
+            let lookup_terms_evals_d1: Evaluations<G::ScalarField, D<G::ScalarField>> =
+                Evaluations::<G::ScalarField, D<G::ScalarField>>::from_vec_and_domain(
+                    lookup_terms_evals,
+                    domain.d1,
+                );
+
+            // FIXME: only one lookup terms for now
+            let lookup_terms_poly_d1: DensePolynomial<G::ScalarField> =
+                lookup_terms_evals_d1.interpolate_by_ref();
+
+            // FIXME: only one lookup terms for now
+            let lookup_terms_comms_d1: PolyComm<G> =
+                srs.commit_evaluations_non_hiding(domain.d1, &lookup_terms_evals_d1);
+
+            // FIXME: only one lookup terms for now
+            absorb_commitment(fq_sponge, &lookup_terms_comms_d1);
+            // -- end computing the row sums h
+
+            // -- start computing the running sum in lookup_aggregation
+            // The running sum, \phi, is defined recursively over the subgroup as followed:
+            // - phi(1) = 0
+            // - phi(\omega^{j + 1}) = \phi(\omega^j) + \
+            //                         \sum_{i = 1}^{n} (1 / \beta + f_i(\omega^{j + 1})) - \
+            //                         (m(\omega^{j + 1}) / beta + t(\omega^{j + 1}))
+            // - phi(\omega^n) = 0
+            let lookup_aggregation_evals_d1 = {
+                let mut evals = Vec::with_capacity(domain.d1.size as usize);
+                let mut acc = G::ScalarField::zero();
+                for i in 0..domain.d1.size as usize {
+                    // phi(1) = 0
+                    evals.push(acc);
+                    // FIXME: only one lookup for now.
+                    acc += lookup_terms_evals_d1[i]
+                }
+                // Sanity check to verify that the accumulator ends up being zero.
+                // FIXME: This should be removed from runtime, and a constraint
+                // should be added. For now, the verifier accepts any proof.
```

Comment:
We don't need a constraint that `acc = 0` in the end, only the one that the whole thing is circular. Can start with any nonzero constant. Maybe this comment is outdated.

Reply:
Yes, it can be anything other than 0. Should be mentioned somewhere.